File: frontend/src/types/reports.ts
export type ReportSummaryItem = {
  behavior: string;
  records: number;
  duration_seconds: number;
};

export type ReportSummaryResponse = {
  items: ReportSummaryItem[];
};

export type BehaviorEpisode = {
  id: number;
  monitoring_session_id?: number | null;
  student_id?: string | null;
  student: string;
  behavior: string;
  start_time: string;
  end_time: string;
  duration_seconds: number;
  date: string;
  source?: string | null;
  lesson_type?: string | null;
  discipline?: string | null;
  teacher?: string | null;
  class_name?: string | null;
  class_identifier?: string | null;
};

export type BehaviorEpisodeListResponse = {
  items: BehaviorEpisode[];
  total: number;
};

File: backend/app/schemas/reports.py
from datetime import date, datetime

from pydantic import BaseModel


class ReportSummaryItem(BaseModel):
    behavior: str
    records: int
    duration_seconds: float


class ReportSummaryResponse(BaseModel):
    items: list[ReportSummaryItem]


class BehaviorEpisodeRead(BaseModel):
    id: int
    monitoring_session_id: int | None = None
    student_id: str | None = None
    student: str
    behavior: str
    start_time: datetime
    end_time: datetime
    duration_seconds: float
    date: date
    source: str | None = None
    lesson_type: str | None = None
    discipline: str | None = None
    teacher: str | None = None
    class_name: str | None = None
    class_identifier: str | None = None


class BehaviorEpisodeListResponse(BaseModel):
    items: list[BehaviorEpisodeRead]
    total: int
